app/src/Blocks/Column/MultiColumnBlock.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+
+namespace SilverStripe\Bambusa\Blocks\Column;
+
+use DNADesign\Elemental\Models\BaseElement;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldAddNewButton;
+use SilverStripe\Forms\GridField\GridFieldConfig_RecordEditor;
+use SilverStripe\Shared\Models\ContentItem;
+use Symbiote\GridFieldExtensions\GridFieldOrderableRows;
+
+class MultiColumnBlock extends BaseElement
+{
+    /**
+     * @var string
+     */
+    private static $icon = 'font-icon-columns';
+
+    /**
+     * @var string
+     */
+    private static $table_name = 'MultiColumnBlock';
+    /**
+     * @var string
+     */
+    private static $singular_name = 'Multi-column';
+
+    /**
+     * @var string
+     */
+    private static $plural_name = 'Multi-column blocks';
+
+    /**
+     * @var string
+     */
+    private static $description = 'Multiple column blocks';
+
+    /**
+     * @var bool
+     */
+    private static $inline_editable = false;
+
+    /**
+     * @var array
+     */
+    private static $has_many = [
+        'Items' => Item::class
+    ];
+
+    /**
+     * @var array
+     */
+    private static $owns = [
+        'Items'
+    ];
+
+    private static $cascade_duplicates = [
+        'Items'
+    ];
+
+    public function getCMSFields(): FieldList
+    {
+        $fields = parent::getCMSFields();
+        $fields->removeByName('Items');
+        $fields->addFieldToTab(
+            'Root.Main',
+            GridField::create(
+                'Items',
+                'Items',
+                $this->Items(),
+                $config = GridFieldConfig_RecordEditor::create()
+            )
+        );
+
+        $config->addComponent(GridFieldOrderableRows::create());
+
+        // Limit columns to 4 per block
```

Comment:
I think this block would be more useful if it was called `GridBlock`, similar to https://github.com/thewebmen/silverstripe-elemental-grid

But it's not a part of the AC's, so feel free to ignore this.

Reply:
Would it be completely correct though? My assumption is that Grid gives you more control over vertical composition of the columns. Here we don't give any control over it - users would have to create another block to start a new row.

Reply:
Yep, it would require us to add the ability to add rows, but it is not worth doing unless we make this a separate module I guess.